src/accelerate/data_loader.py

```diff
@@ -382,7 +383,18 @@ def __iter__(self):
             # Broadcast the batch before splitting it.
             batch = broadcast(batch, from_process=0)
 
-            batch_size = find_batch_size(batch) // state.num_processes
+            if not self.drop_last and first_batch is None:
+                # We keep at least num processes elements of the first batch to be able to complete the last batch
+                first_batch = slice_tensors(batch, slice(0, state.num_processes))
+
+            observed_batch_size = find_batch_size(batch)
+            batch_size = observed_batch_size // state.num_processes
+
+            if not self.drop_last and stop_iteration and observed_batch_size % state.num_processes != 0:
+                # If the last batch is not complete, let's add the first batch to it.
+                batch = concatenate([batch, first_batch], dim=0)
```

Comment:
Will this run twice on the first batch in case `drop_last` is `False`?

Reply:
Yes, like all the dataloading utils, the batches need to be the same length on all processes, which means we have to add some data from the beginning to completet the last batch when `drop_last=False`.